pkg/sdk/logging/model/syslogng/output/s3.go

```diff
@@ -0,0 +1,104 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package output
+
+import "github.com/cisco-open/operator-tools/pkg/secret"
+
+// +name:"S3"
+// +weight:"200"
+type _hugoS3 interface{} //nolint:deadcode,unused
+
+// +docName:"Sending messages from a local network to a S3 (compatible) server"
+/*
+## Example
+{{< highlight yaml >}}
+apiVersion: logging.banzaicloud.io/v1beta1
+kind: SyslogNGOutput
+metadata:
+name: s3
+spec:
+s3:
+  url: "https://some-s3-compatible-endpoint:8088"
+  bucket: "s3bucket-name"
+  access_key:
+    valueFrom:
+      secretKeyRef:
+        name: s3
+        key: access-key
+  secret_key:
+    valueFrom:
+      secretKeyRef:
+        name: s3
+        key: secret-key
+  object_key:
+    valueFrom:
+      secretKeyRef:
+        name: s3
+        key: object-key
+{{</ highlight >}}
+*/
+type _docS3 interface{} //nolint:deadcode,unused
+
+// +name:"S3 Destination"
+// +url:"https://axoflow.com/docs/axosyslog-core/chapter-destinations/destination-s3/"
+// +description:"Sending messages from a local network to a S3 (compatible) server"
+// +status:"Testing"
+type _metaS3 interface{} //nolint:deadcode,unused
+
+// +kubebuilder:object:generate=true
+type S3Output struct {
+	// The hostname or IP address of the S3 server.
+	Url string `json:"url,omitempty"`
+	// The bucket name of the S3 server.
+	Bucket string `json:"bucket,omitempty"`
+	// The access_key for the S3 server.
+	AccessKey *secret.Secret `json:"access_key,omitempty"`
+	// The secret_key for the S3 server.
+	SecretKey *secret.Secret `json:"secret_key,omitempty"`
+	// The object_key for the S3 server.
+	ObjectKey *secret.Secret `json:"object_key,omitempty"`
+	// Set object_key_timestamp
+	ObjectKeyTimestamp RawString `json:"object_key_timestamp,omitempty"`
+	// Template
```

Comment:
Nit: for this and other syslog-ng specific fields, it may be useful to reference the syslog-ng docs.

Reply:
We have a doc link for the output in general (which is not yet released) so I think this is fine now.